File: HFSGuardaDiretorio_SharpDevelop_Ruby/gui/FrmImportarDiretorioProgresso.rb
# 
# * Created by SharpDevelop.
# * User: henrique.souza
# * Date: 08/07/2015
# * Time: 11:26
# *
# * To change this template use Tools | Options | Coding | Edit Standard Headers.
# 
require "mscorlib"
require "System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "HFSGuardaDiretorio.comum, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

module HFSGuardaDiretorio.gui
	# <summary>
	# Description of FrmImportarDiretorioProgresso.
	# </summary>
	class FrmImportarDiretorioProgresso < IProgressoLog
		def initialize(form)
			self.@form = form
		end

		def ProgressoLog(progresso)
			if progresso.Posicao == 0 then
				self.@form.pbImportar.Minimum = progresso.Minimo
				self.@form.pbImportar.Maximum = progresso.Maximo
				self.@form.pbImportar.Step = progresso.Passo
			end
			self.@form.pbImportar.Value = progresso.Posicao
			if progresso.Log.Length > 0 then
				pos = @form.memoImportaDir.TextLength
				@form.memoImportaDir.AppendText(progresso.Log + "\n")
				@form.memoImportaDir.Select(pos, pos)
				@form.barraStatus.Items[1].Text = Convert.ToString(pos)
				@form.barraStatus.Update()
			end
		end
	end
end

File: HFSGuardaDiretorio_SharpDevelop_Python/gui/FrmImportarDiretorioProgresso.py
# 
# * Created by SharpDevelop.
# * User: henrique.souza
# * Date: 08/07/2015
# * Time: 11:26
# *
# * To change this template use Tools | Options | Coding | Edit Standard Headers.
# 
from System import *
from HFSGuardaDiretorio.comum import *

class FrmImportarDiretorioProgresso(IProgressoLog):
	""" <summary>
	 Description of FrmImportarDiretorioProgresso.
	 </summary>
	"""
	def __init__(self, form):
		self._form = form

	def ProgressoLog(self, progresso):
		if progresso.Posicao == 0:
			self._form.pbImportar.Minimum = progresso.Minimo
			self._form.pbImportar.Maximum = progresso.Maximo
			self._form.pbImportar.Step = progresso.Passo
		self._form.pbImportar.Value = progresso.Posicao
		if progresso.Log.Length > 0:
			pos = self._form.memoImportaDir.TextLength
			self._form.memoImportaDir.AppendText(progresso.Log + "\n")
			self._form.memoImportaDir.Select(pos, pos)
			self._form.barraStatus.Items[1].Text = Convert.ToString(pos)
			self._form.barraStatus.Update()